flax/linen/summary.py

```diff
@@ -114,222 +121,200 @@ def __call__(self, x):
 
   Returns:
     A function that accepts the same `*args` and `**kwargs` of the forward pass
     (`method`) and returns a string with a tabular representation of the
     Modules.
   """
 
-  def _tabulate_fn(*args, **kwargs):
-    table_fn = _get_module_table(module, rngs, method=method, 
-                                 mutable=mutable, depth=depth, 
-                                 exclude_methods=set(exclude_methods))
-    table = table_fn(*args, **kwargs)
-    return _render_table(table)
+  def _tabulate_fn(*fn_args, **fn_kwargs):
+    table_fn = _get_module_table(module, depth=depth, show_repeated=show_repeated)
+    table = table_fn(rngs, *fn_args, mutable=mutable, **fn_kwargs, **kwargs)
+    return _render_table(table, console_kwargs)
 
   return _tabulate_fn
 
+# TODO(cgarciae): Remove this when paths are fixed
+# fix path logic where lifted modules paths don't match the variable structure
+# when custom names are used.
+_find_non_lifted_module = re.compile(r'.*\((.*)\)')
+def _fix_path_part(part: str):
+  match = _find_non_lifted_module.match(part)
+  if match:
+    return match.group(1)
+  return part
+
+def _fix_call_info_path(call_info: 'flax.linen.module._CallInfo') -> 'flax.linen.module._CallInfo':
+  from flax.linen.module import _CallInfo
+  path = tuple(
+      _fix_path_part(part) for part in call_info.path
+  )
+  return _CallInfo(call_info.index, path, *call_info[2:])
+
 
 def _get_module_table(
     module: 'flax.linen.Module',
-    rngs: Union[PRNGKey, RNGSequences],
-    method: Optional[Callable[..., Any]],
-    mutable: CollectionFilter,
     depth: Optional[int],
-    exclude_methods: Set[str],
+    show_repeated: bool,
 ) -> Callable[..., Table]:
-
-  exclude_methods.add("setup")
+  import flax.linen.module as module_lib
 
   def _get_table_fn(*args, **kwargs):
-    output_methods: Set[str] = set()
-
-    def capture_intermediates(_module, method_name: str):
-      if method_name in exclude_methods:
-        return False
+    
+    variables = FrozenDict()
+    calls = []
+
+    with module_lib._call_info_context.tabulate_context():
+      assert module_lib._call_info_context.calls is not None
+
+      def _get_variables():
+        variables = module.init(*args, **kwargs)
+        return variables
+
+      variables = jax.eval_shape(_get_variables)
+      calls = module_lib._call_info_context.calls
+      calls.sort(key=lambda c: c.index)
+      # fix path
+      calls = [_fix_call_info_path(c) for c in calls]
+
+    collections: Set[str] = set(variables.keys())
+    rows = []
+    all_paths: Set[Tuple[str, ...]] = set(call.path for call in calls)
+    visited_paths: Set[Tuple[str, ...]] = set()
+
+    for i, path, module_type, method, call_args, call_kwargs, outputs in calls:
+      call_depth = len(path)
+      inputs = _process_inputs(call_args, call_kwargs)
+
+      if path in visited_paths:
+        if not show_repeated:
+          continue
+        module_vars = {}
+        counted_vars = {}
```

Comment:
depth is not checked if show_repeated=True?

Reply:
`path in visited_paths` is never true the first time around so depth is always taken into account.